src/main/java/io/debezium/connector/jdbc/ReducedRecordBuffer.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.jdbc;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.ConnectException;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.List;
+import java.util.Objects;
+import java.util.HashMap;
+
+/**
+ * A reduced implementation buffer of {@link SinkRecordDescriptor}.
+ * It reduces events in buffer before submit to external database.
+ *
+ * @author Gaurav Miglani
+ */
+public class ReducedRecordBuffer implements Buffer {
+
+    private final JdbcSinkConnectorConfig connectorConfig;
+    private Schema keySchema;
+    private Schema valueSchema;
+
+    private final Map<Struct, SinkRecordDescriptor> records = new HashMap<>();
+
+    public ReducedRecordBuffer(JdbcSinkConnectorConfig connectorConfig) {
+        this.connectorConfig = connectorConfig;
+    }
+
+    @Override
+    public List<SinkRecordDescriptor> add(SinkRecordDescriptor recordDescriptor) {
+        ArrayList<SinkRecordDescriptor> flushed = new ArrayList<>();
```

Comment:
I see two issues here

1) The type should be `List`
2) Is there a way how to change the code so we can avoid copying the content of the buffer into the temporary variable? Ideally the result of method `flush()` should be returned directly. BTW I don't have problem if buffer size limit is temporarily broken in case of PK struct change to simplify that perf improvement

Reply:
got it, working on it @jpechane

Reply:
i have created a boolean variable `isSchemaChanged` to solve copying issue, raise a commit, please review @jpechane @Naros

Reply:
also @jpechane , the initial logic copying the content of the buffer into the temporary variable, but this is soft copy , keeping pointers only for `SinkRecordDescriptor` object , not exact `SinkRecordDescriptor` object copy, correct ? I'm not sure, how much perf improvement will be there.

Reply:
Also please bear in mind that we don't require single `return` from method so the code might at the end be easier to implement/comprehend if multiple returns are used.

Reply:
I have made the changes using variable `isSchemaChanged`, is it correct @jpechane  ?

Reply:
@jpechane waiting for your review